perception/lidar_centerpoint/lib/preprocess/voxel_generator.cpp

```diff
@@ -38,35 +40,30 @@ VoxelGeneratorTemplate::VoxelGeneratorTemplate(
 }
 
 bool VoxelGeneratorTemplate::enqueuePointCloud(
-  const sensor_msgs::msg::PointCloud2 & input_pointcloud_msg, const tf2_ros::Buffer & tf_buffer)
+  const sensor_msgs::msg::PointCloud2 & input_pointcloud_msg, const tf2_ros::Buffer & tf_buffer,
+  cudaStream_t stream)
 {
-  return pd_ptr_->enqueuePointCloud(input_pointcloud_msg, tf_buffer);
+  return pd_ptr_->enqueuePointCloud(input_pointcloud_msg, tf_buffer, stream);
 }
 
-std::size_t VoxelGenerator::generateSweepPoints(std::vector<float> & points)
+std::size_t VoxelGenerator::generateSweepPoints(float * points_d, cudaStream_t stream)
 {
-  Eigen::Vector3f point_current, point_past;
-  size_t point_counter{};
+  size_t point_counter = 0;
   for (auto pc_cache_iter = pd_ptr_->getPointCloudCacheIter(); !pd_ptr_->isCacheEnd(pc_cache_iter);
```

Comment:
```
[lidar_centerpoint_node-1]   what():  cudaErrorIllegalAddress (700)@/home/amadeusz/autoware/src/autoware/universe/perception/lidar_centerpoint/include/lidar_centerpoint/cuda_utils.hpp#L68: an illegal memory access was encountered
```
With large input cloud, array may exceed limit. FYI you can observe it if you increase cloud cache size. I think here we can do one of options:
* Check if current `point_counter` + `pc_cache_iter->num_points` is within the limit of `CAPACITY_POINT`. If not, break loop.
* Pass `CAPACITY_POINT` to kernel and monitor current size. If `CAPACITY_POINT` is hit, return.

I'm for the 1st option. With 2nd options we will get one benefit in form of unknown part of cloud, but extra operations in kernel will be necessary.

Reply:
Thanks for the catch!
Addressed in [80112b1](https://github.com/autowarefoundation/autoware.universe/pull/6989/commits/80112b1be14753f94cde12910e28d1b7fda91366)